pkg/timeline/adaptor/repository/dummyCache.ts

```diff
@@ -20,15 +21,41 @@ export class InMemoryTimelineCacheRepository
     );
   }
 
-  private generateObjectKey(accountID: AccountID): CacheObjectKey {
-    return `timeline:home:${accountID}`;
+  private generateObjectKey(
+    id: AccountID | ListID,
+    timelineType: 'home' | 'list',
+  ): CacheObjectKey {
+    return `timeline:${timelineType}:${id}`;
   }
 
   async addNotesToHomeTimeline(
     accountID: AccountID,
     notes: Note[],
   ): Promise<Result.Result<Error, void>> {
-    const objectKey = this.generateObjectKey(accountID);
+    const objectKey = this.generateObjectKey(accountID, 'home');
+    if (!this.data.has(objectKey)) {
+      this.data.set(
+        objectKey,
+        notes.map((note) => note.getID()),
+      );
+      return Result.ok(undefined);
+    }
+    // biome-ignore lint/style/noNonNullAssertion: Use assertion to avoid compile errors because type inference has failed.
+    const fetched = this.data.get(objectKey)!;
+    // NOTE: replace by updated object
+    this.data.delete(objectKey);
+
+    fetched.push(...notes.map((note) => note.getID()));
+    this.data.set(objectKey, fetched);
+
+    return Result.ok(undefined);
```

Comment:
```suggestion
    const newNoteIds = notes.map((note) => note.getID());
    if (!this.data.has(objectKey)) {
      this.data.set(objectKey, newNoteIds);
      return Result.ok(undefined);
    }
    this.data.get(objectKey)?.push(...newNoteIds);
    return Result.ok(undefined);
```
`objectKey` のエントリに `Note` の ID を追加するだけならこれでもよさそうです.